docs/basic/complexity.md

```diff
@@ -1,60 +1,155 @@
 
-### 大 Θ 符号
+The count or estimation of elementary operations can be criteria when evaluating the efficiency of an algorithm. 
 
-对于给定的一个函数 $g(n)$，$f(n)=\Theta(g(n))$，当且仅当 $\exists c_1,c_2,n_0>0$，使得 $\forall n \ge n_0, 0\le c_1\cdot g(n)\le f(n) \le c_2\cdot g(n)$。
+## Time Complexity
 
-也就是说，如果函数 $f(n)=\Theta(g(n))$，那么我们能找到两个正数 $c_1, c_2$ 使得 $f(n)$ 被 $c_1\cdot g(n)$ 和 $c_2\cdot g(n)$ 夹在中间。
+To evaluate the performance of an algorithm, it is required to take the data scale<!---「数据规模」是这么翻译的喵？-->, which mostly referring to the counts of numbers inputted, the counts of points or edges of a graph inputted, into consideration. Generally, the larger the data scale, the longer the algorithm performs. But when we evaluate the performance in competitive programming, the most important is not its time cost in a specific data scale, but its trend with the data scale growing, as known as **time complexity**.
 
-### 大 O 符号
+The main reasons for considering in this way are listed below:
 
-$\Theta$ 符号同时给了我们一个函数的上下界，如果我们只有一个函数的渐进上界的时候，我们使用 $O$ 符号。对于一个给定的函数 $g(n)$, 我们把它记作 $O(g(n))$。$f(n)=O(g(n))$，当且仅当 $\exists c,n_0$，使得 $\forall n \ge n_0,0\le f(n)\le c\cdot g(n)$。
+1. Modern computers are capable to process billions or more elemental operations, so the data scale we need to process is usually very large. For example, assume that algorithm A has a time cost of $100n$ for proceeding data of $n$ size, and algorithm B's time cost is $n^2$. Algorithm B takes less time when the data scale is less than $100$. But in one second algorithm A can proceed data on the scale of millions, while algorithm B only on the scale of thousands. Thus, with the longer acceptable processing time, time complexity has far more obvious impact on the scale of processable data than the impact of time cost at the same data scale.
+2. We use the counts of elementary operations to represent the time cost of an algorithm. However, different kinds of elementary operations' time cost differs, for example, time cost of addition and subtraction is far less than division. By ignoring the difference between different kinds and counts of operations when calculating time complexity, we can eliminate the impact of different time cost between operations.
 
-研究时间复杂度时通常会使用 $O$ 符号，因为我们关注的通常是程序用时的上界，而不关心其用时的下界。
+Of course, the running time of an algorithm is not entirely determined by the input scale, but is also correlated with the content of the input. Therefore, the time complexity is further divided into several categories. For example:
 
-### 大 Ω 符号
+1. Worst-case time complexity, which the largest input scale corresponds for each input scale. In competitive programming, since the input can be given arbitrarily in given data scale, to ensure an algorithm can proceed any data in some scale in time, we generally take worst-case time complexity into consideration.
```

Comment:
```suggestion
1. Worst-case time complexity, which the largest input scale corresponds for each input scale. In competitive programming, since the input can be given arbitrarily under given constraint, to ensure an algorithm can process any data under constraints given within time limitation, we generally take worst-case time complexity into consideration.
```
🤔 感觉 data scale 和 constraint 还是有区别的？constraint（数据范围）感觉还可以有 `n=1` 这种

Reply:
they use *size*.

显然wikipedia里没有data scale/constraint的

恩，全改掉。